src/app/InteractionModelEngine.h

```diff
@@ -132,6 +133,21 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
      */
     CASESessionManager * GetCASESessionManager() const { return mpCASESessionMgr; }
 
+    // reporting::EngineDelegate implementation
+    Messaging::ExchangeManager * GetExchangeManagerForReportingEngine() override { return mpExchangeMgr; }
+    ObjectPool<ReadHandler, CHIP_IM_MAX_NUM_READS + CHIP_IM_MAX_NUM_SUBSCRIPTIONS> & GetReadHandlerPool() override
+    {
+        return mReadHandlers;
+    }
+    ReadHandler * ActiveHandlerAt(unsigned int aIndex) override;
+    reporting::ReportScheduler * GetReportScheduler() override { return mReportScheduler; }
```

Comment:
Raising the question of whether this should even be on the IM engine at all.

Reply:
I think some design choices should be updated here ... however for now I just mechanically moved things around.

I will take a 2nd pass to make things slightly better, however unsure if I will be able to also move this around as part of this work or not.

In my mind, app/reporting seems to be a stand-alone folder and should probably be a stand-alone library, so all this coupling is somewhat suspect.